integration/tests/epochs/suite.go

```diff
@@ -96,3 +101,246 @@ func (s *Suite) TearDownTest() {
+// 4. Add additional funds to staking account for storage
+// 5. Create Staking collection for node
+// 6. Register node using staking collection object
+func (s *Suite) StakeNode(role flow.Role) {
+	stakingAccountKey, networkingKey, stakingKey, _, machineAccountPubKey := s.generateAccountKeys()
+	nodeID := flow.MakeID(stakingKey.PublicKey().Encode())
+	fullAccountKey := sdk.NewAccountKey().
+		SetPublicKey(stakingAccountKey.PublicKey()).
+		SetHashAlgo(sdkcrypto.SHA2_256).
+		SetWeight(sdk.AccountKeyWeightThreshold)
+
+	// create staking account
+	stakingAccountAddress, err := s.createNewLeaseAccount(fullAccountKey)
+	require.NoError(s.T(), err)
+
+	// fund account with token amount to stake
+	result, err := s.transferLeaseTokens(role, stakingAccountAddress)
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	// fund account for storage
+	result, err = s.fundAccount(stakingAccountAddress, "10.0")
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	stakingAccount, err := s.client.GetAccount(stakingAccountAddress)
+	require.NoError(s.T(), err)
+
+	// create staking collection
+	_, err = s.createStakingCollection(stakingAccountKey, stakingAccount)
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	tokenAmount, err := s.client.TokenAmountByRole(flow.RoleConsensus.String())
+	require.NoError(s.T(), err)
+
+	encMachinePubKey, err := flow.EncodeRuntimeAccountPublicKey(machineAccountPubKey)
+	require.NoError(s.T(), err)
+	hex.EncodeToString(encMachinePubKey)
+
+	// register node using staking collection
+	result, err = s.registerNode(
+		stakingAccountKey,
+		stakingAccount,
+		nodeID,
+		flow.RoleConsensus,
+		"localhost:9000",
+		networkingKey.PublicKey().String()[2:],
+		stakingKey.PublicKey().String()[2:],
+		tokenAmount,
+		hex.EncodeToString(encMachinePubKey),
+	)
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+}
+
+// transfers tokens to receiver from service account
+func (s *Suite) fundAccount(receiver sdk.Address, tokenAmount string) (*sdk.TransactionResult, error) {
+	ctx := context.Background()
+	latestBlockID, err := s.client.GetLatestBlockID(ctx)
+	require.NoError(s.T(), err)
+
+	env := utils.EnvFromNetwork("localnet")
+	transferTx, err := utils.MakeTransferToken(
+		env,
+		receiver,
+		s.client.Account(),
+		0,
+		tokenAmount,
+		sdk.Identifier(latestBlockID),
+	)
+	require.NoError(s.T(), err)
+
+	err = s.client.SignAndSendTransaction(ctx, transferTx)
+	require.NoError(s.T(), err)
+
+	result, err := s.client.WaitForSealed(ctx, transferTx.ID())
+	require.NoError(s.T(), err)
+
+	return result, nil
+}
+
+// generates inital keys needed to bootstrap account
+func (s *Suite) generateAccountKeys() (
+	stakingAccountKey,
+	networkingKey,
+	stakingKey,
+	machineAccountKey sdkcrypto.PrivateKey,
+	machineAccountPubKey flow.AccountPublicKey,
+) {
+	stakingAccountKey, err := unittest.NetworkingKey()
+
+	machineAccountKey, err = unittest.NetworkingKey()
```

Comment:
This is a bit confusing, mabye we could do:
```go
func NetworkingKey() { return ecdsaKey() }
func MachineAccountKey() { return ecdsaKey() }
func ecdsaKey() {
  // what NetworkingKey currently does
}
```

Reply:
https://github.com/onflow/flow-go/pull/1320/commits/4b1a90ffb551beb5f4f9cbb833424a62bf9e10b0